cryptoki/src/types/mod.rs

```diff
@@ -699,3 +974,103 @@ impl TryFrom<usize> for Ulong {
         })
     }
 }
+
+impl std::fmt::Display for Ulong {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.val)
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+/// Represents a version
+pub struct Version {
+    major: CK_BYTE,
+    minor: CK_BYTE,
+}
+
+impl std::fmt::Display for Version {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}.{}", self.major, self.minor)
+    }
+}
+
+impl Version {
+    /// Returns the major version
+    pub fn get_major(&self) -> CK_BYTE {
+        self.major
+    }
+
+    /// Returns the minor version
+    pub fn get_minor(&self) -> CK_BYTE {
+        self.minor
+    }
+}
+
+impl From<Version> for CK_VERSION {
+    fn from(version: Version) -> Self {
+        CK_VERSION {
+            major: version.major,
+            minor: version.minor,
+        }
+    }
+}
+
+impl From<CK_VERSION> for Version {
+    fn from(version: CK_VERSION) -> Self {
+        Version {
+            major: version.major,
+            minor: version.minor,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+/// Type identifying the PKCS#11 library information
+pub struct Info {
+    val: CK_INFO,
+}
+
+impl Info {
+    pub(crate) fn new(val: CK_INFO) -> Self {
+        Self { val }
+    }
+
+    /// Returns the version of Cryptoki that the library is compatible with
+    pub fn get_cryptoki_version(&self) -> Version {
```

Comment:
Getters don't usually have the `get_` in front

Reply:
Ah, that's probably my `C++` based brain :)